sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedParquetRecordReader.java

```diff
@@ -415,4 +417,36 @@ private void initColumnReader(PageReadStore pages, ParquetColumnVector cv) throw
       }
     }
   }
+
+  /**
+   * <b>This method assumes that all constant column are at the end of schema
+   * and `constantColumnLength` represents the number of constant column.<b/>
+   *
+   * This method allocates columns to store elements of each field of the schema,
+   * the data columns use `OffHeapColumnVector` when `useOffHeap` is true and
+   * use `OnHeapColumnVector` when `useOffHeap` is false, the constant columns
+   * always use `ConstantColumnVector`.
+   *
+   * Capacity is the initial capacity of the vector, and it will grow as necessary.
+   * Capacity is in number of elements, not number of bytes.
+   */
+  private ColumnVector[] allocateColumns(
```

Comment:
nit: we can make it static?

Reply:
Yes, but what are the benefits of static? I didn't think clearly

Reply:
Yea, it doesn't matter much in this case, mostly just a code style thing. I'm fine with `private` too.

Reply:
OK ~